rfc/cdevents-spinnaker.md

```diff
@@ -0,0 +1,106 @@
+
+
+An `events-broker` will be created to send and receive events by knative-eventing and the sample URL can be configured as below in the cluster.
+
+`events-broker : http://broker-ingress.knative-eventing.svc.cluster.local/default/events-broker`
+
+#### Consume CDEvents
+
+Spinnaker API endpoint `cdevents/consume` will be subscribed to `events-broker` to listen to the various types of events and Spinnaker Pipelines need to be configured with the `CDEvents` type on which Spinnaker is interested on.
+Example: A sample event `dev.cdevents.artifact.published` subscribed to `events-broker`
+``` 
+kubectl create -f - <<EOF
+apiVersion: eventing.knative.dev/v1
+kind: Trigger
+metadata:
+  name: cd-artifact-published-to-spinnaker
+spec:
+  broker: events-broker
+  filter:
+    attributes:
+      type: dev.cdevents.artifact.published
+  subscriber:
+    uri: http://<spin-service>:<port>/cdevents/consume
+EOF 
+```
+Once the event is received by Spinnaker API endpoint `cdevents/consume`, further can be handled to run pipelines based on the configured event data in Spinnaker.
+
+#### Produce CDEvents
+Spinnaker API endpoint `cdevents/produce` will be invoked with the type of event that needs to be sent to `events-broker`.
+`CDEvents` can be created using [Java SDK](https://github.com/cdevents/sdk-java) that is available in `CDEvents` [GitHub Repositories](https://github.com/cdevents).
+`Java SDK` used to create variuos types of events alligning with `CDEvents` vocabulary specification using CloudEventBuilder. 
+
+`CDEvents` can be published to `events-broker` once created from Spinnaker application using CloudEvents HTTP libraries.
+
+Example: A sample serivce-deployed event published to `events-broker`
+```
+curl -v -d '{"id": "1234", "subject": "event"}' -X POST -H "Ce-Id: HelloSpinnaker" -H "Ce-Specversion: 1.0" -H "Ce-Type: dev.cdevents.service.deployed" -H "Ce-Source: not-sendoff" -H "Content-Type: application/json" "http://<events-broker-url>/default/events-broker"
+```
+
+### Dependencies
+
+The CloudEvent dependencies(io.cloudevents) are required to implement produce/consume events 
+
+
+## Prior Art and Alternatives
+The existing approach is integration oriented and could sometimes result in outages due to change in implementation of integrated components. `CDEvents` allows users to take a more standardised approach, ensuring efficiency and sustainability while creating and maintaining CI/CD pipelines.
+
+## Known Unknowns
+
+The approach will be demonstrated with the Spinnaker Community using `CDEvents` PoC.
+
+## Operations
+Admin needs to configure Pipelines that should run on consuming the different event types from `events-broker`
+
+## Future Possibilities
+
+`CDEvents` open the doors to scalable and decoupled interoperability within the software supply chain and create the potential for greater visibility and measurability of cloud native continuous delivery workflows.
```

Comment:
SO overall like the ideas here, what I'd like is a bit more depth both on:
* Implementation, e.g. how it'll actually flow, who's going to help maintain it going forward (aka a SIG for this?), and some details
* Concrete other providers that we can reference for docs/implementation details/etc.
* Impacts long term and security details.  E.g. what permissions would consumers use?  How could they be restricted from getting ALL events to only some events?  Etc.?

Reply:
+1 on that. This is way too high level. We need to figure out how to ensure the context remains the same if we do decide on an event bus.